packages/react-storage/src/components/StorageBrowser/Views/Controls/Loading.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+import type { OmitElements } from '../types';
+import { StorageBrowserElements } from '../../context/elements';
+import { CLASS_BASE } from '../constants';
+
+const { Icon, Span, View } = StorageBrowserElements;
+
+const BLOCK_NAME = `${CLASS_BASE}__loading`;
+
+interface LoadingControlProps {
+  loadingText?: string | undefined;
+}
+
+export interface _LoadingControl<
+  T extends StorageBrowserElements = StorageBrowserElements,
+> extends Pick<T, 'Icon' | 'Span' | 'View'> {
+  ({ loadingText }: LoadingControlProps): React.JSX.Element;
+}
+
+export interface LoadingControl<
+  T extends StorageBrowserElements = StorageBrowserElements,
+> extends OmitElements<_LoadingControl<T>, 'Icon' | 'Text' | 'View'> {
+  ({ loadingText }: LoadingControlProps): React.JSX.Element;
+}
+
+export const LoadingControl: LoadingControl = ({ loadingText }) => (
+  <View className={BLOCK_NAME}>
```

Comment:
It is possible to use a fragment here and use flex properties to center the icon and expand the margins of the Icon to fill the parent container?

Reply:
I think that would make the layout of this component a little too dependent on the parent; and a little more brittle should you change any of the styling on the parent (which in this case would be the top level `.storage-browser` and it's grid/alignment styling). I'd prefer to keep the internal layout of this (the flex positioning, centering) dependent on it's own wrapper. That way if developers want to change the internal layout, like stacking the icon on top of the text for mobile, adding a background/padding, etc. it will be easier to do.